autosklearn/ensemble_builder.py

```diff
@@ -229,7 +227,7 @@ def ensemble_builder_process(
 
         # If pynisher kills a run, the result
         # might be None -- so no new timestamp info
-        if future.done():
+        if future is not None and future.done():
```

Comment:
Could we actually wait for the future to be done? Maybe with a timeout of 1s to also check for the ensemble termination request? I believe this would be closer to how dask actually wants things to be done.